### PR TITLE
[fix]: [CDS-3]: K8s & AWS error message improvements

### PR DESCRIPTION
**Pull request by abhinavsingh1196**
Improvements:
- AWS connector validation message improvements & wrapped NPE error message in WingsException
- K8s error message improvement when multiple invalid resources are provided
- K8s error handling for missing/null field in manifest

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30851)

### CODE REVIEW

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30851)* status: Some of the checks are too old.

File: 980-commons/src/main/java/io/harness/exception/ExplanationException.java [hunk 1]
```diff
@@ -18,16 +18,16 @@
...
   public static final String EXPLANATION_EMPTY_ACCESS_KEY = "Access Key cannot be empty";
   public static String EXPLANATION_EMPTY_SECRET_KEY = "Secret Key cannot be empty";
   public static String EXPLANATION_EMPTY_CROSS_ACCOUNT_ROLE_ARN_KEY = "Cross account role ARN cannot be empty";
   public static String EXPLANATION_INCORRECT_ACCESS_KEY_SECRET_KEY_PERMISSIONS_KEY =
-      "Access Key or Secret Key is invalid. User may not have required permissions to perform the activity.";
+      "Access Key or Secret Key is invalid. User may not have required permissions to describe regions.";
   public static String EXPLANATION_INVALID_CROSS_ACCOUNT_ROLE_ARN_EXTERNAL_ID_PERMISSIONS_KEY =
-      "Cross account role ARN or External Id is invalid or User / IAM Role on delegate / IRSA role doesn't have permission to perform sts:AssumeRole. Assumed Cross account role may not have permissions to perform the activity.";
+      "\nCross account role ARN or External Id is invalid or User / IAM Role on delegate / IRSA role doesn't have permission to perform sts:AssumeRole. \nAssumed Cross account role may not have permissions to perform the activity.";
```
Comment (sainath-b) [inline 980-commons/src/main/java/io/harness/exception/ExplanationException.java:28]: Line 28: 'perform the activity' need to be replaced with describe regions.

File: 980-commons/src/main/java/io/harness/exception/HintException.java [hunk 2]
```diff
@@ -18,15 +18,21 @@
...
   public static final String HINT_INCORRECT_ACCESS_KEY_SECRET_KEY_PERMISSIONS_KEY =
-      "Check if access key, secret key are valid. Check if user has required permissions to perform the activity.";
+      "Check if access key, secret key are valid. \nCheck if user has required permissions to describe regions.";
   public static final String HINT_INVALID_CROSS_ACCOUNT_ROLE_ARN_EXTERNAL_ID_PERMISSIONS_KEY =
-      "Check if Cross account role ARN, External Id are valid. Check if User / IAM Role on delegate / IRSA role has permission to perform sts:AssumeRole. Check if assumed Cross account role has permissions to perform the activity.";
-  public static final String HINT_AWS_IAM_ROLE_CHECK = "Check IAM role on delegate ec2.";
+      "\nCheck if Cross account role ARN, External Id are valid. \nCheck if User / IAM Role on delegate / IRSA role has permission to perform sts:AssumeRole. \nCheck if assumed Cross account role has permissions to describe regions.";
+  public static final String HINT_AWS_IAM_ROLE_CHECK =
+      "Check IAM role on delegate ec2 has the permissions to describe regions in the target account.";
```
[the remark this answers is not in the document]
Comment (sainath-b) [reply 980-commons/src/main/java/io/harness/exception/HintException.java:31]: 'delegate ec2' to 'ec2 delegate'. 'in the target account' can be removed here as well.

**Review by sainath-b — APPROVED** [review]

**Review by sainath-b — APPROVED** [review]
> Reviewed 5 of 7 files at r1, 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30851)* status: :shipit: complete! ✓

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30851)* status: :shipit: complete! ✓

**Pull request merged**